scipy/optimize/linprog_ip.py

```diff
@@ -0,0 +1,1994 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Feb 02 17:09:17 2017
+
+@author: Matt
+
+To do:
+multiple presolve
+questions for scipy-dev
+* This is written in Python 2. How should I check compatibility with Python 3?
+* Are the warnings OK? Do I need an option to disable them?
+* Is 100% pep8 compliance necessary?
+* How should unit tests be performed on both simplex and ip?
+* Can I leave multi-pass presolve to an update?
+* Can I leave sparse presolve to an update?
+* Do I need to implement a callback interface?
+* There are some changes to be made to simplex; it should use the ip presolve
+ routine. Can that be left to an update?
+* How do I document options not common with simplex?
```

Comment:
I wrote about the scope of linear programming methods to the scipy-dev
list, including a bit about the different interior-point methods:

"The ellipsoidal algorithm, Karmarkar's projective algorithm, and affine
scaling method are a few names of historical importance, but I've seen
several sources (not just Wikipedia...) state that primal-dual
path-following algorithms are the preferred choice today."

So, while there *are* different interior point methods, the only one that
seems really important at present is the primal-dual path-following method,
the one I implemented. Based on this, I think "interior-point" is
sufficiently descriptive as it is unlikely that there will be a need for,
say, the ellipsoidal algorithm in SciPy.

Of course, there are certainly variants of primal-dual path following, but
I think `options`, rather than `method`, could be used to switch among them.